plugins/org.locationtech.udig.project.ui/src/org/locationtech/udig/project/ui/internal/commands/draw/DrawCoordinateCommand.java

```diff
@@ -0,0 +1,244 @@
+
+    private Color fill;
+
+    private boolean useCircle;
+    
+    private boolean errorReported;
+    
+    
+    /**
+     * @param coord
+     * @param map Map to consider as active for drawing
+     */
+    public DrawCoordinateCommand(Coordinate coord, IMap map) {
+        this(coord, map.getEditManager().getSelectedLayer().getCRS());
+        setMap(map);
+    }
+
+    /**
+     * @param coord
+     * @param crs
+     */
+    public DrawCoordinateCommand(Coordinate coord, CoordinateReferenceSystem crs) {
+        this.coord = coord;
+        if (crs == null) {
+            this.crs = DefaultGeographicCRS.WGS84;
+        } else {
+            this.crs = crs;
+        }
+        useCircle = false;
+    }
+
+    /**
+     * Does the actual job by drawing a 3x3 rectangle or circle at the 
+     * coordinate location
+     * 
+     * @see org.locationtech.udig.project.command.MapCommand#run()
+     */
+    public void run(IProgressMonitor monitor) throws Exception {
+
+        final Point point = getMap().getViewportModel().worldToPixel(coord);
+        Shape shape = useCircle ? new Ellipse2D.Double(point.x, point.y, 3, 3) : 
+            new Rectangle2D.Double(point.x, point.y, 3, 3);
+        
+        if (fill != null) {
+            graphics.setColor(fill);
+            graphics.fill(shape);
+        }
+
+        if (paint != null)
+            graphics.setColor(paint);
+        if (style > -1)
+            graphics.setStroke(style, width);
+        doDraw(shape);
+    }
+
+    /**
+     * actually the Shape is always of type Rectangle2D but we may enhance 
+     * and make it configurable in the future.
+     *  
+     * @param shape
+     */
+    private void doDraw(Shape shape) {
+        if (shape instanceof Rectangle2D) {
+            Rectangle2D rect = (Rectangle2D) shape;
+            graphics.drawRect((int) rect.getX(), (int) rect.getY(), (int) rect.getWidth(),
+                    (int) rect.getHeight());
+            return;
+        }
+
+        if (shape instanceof Ellipse2D) {
+            Ellipse2D ellipse = (Ellipse2D) shape;
+            graphics.drawOval((int) ellipse.getMinX(), (int) ellipse.getMinY(),
+                    (int) ellipse.getWidth(), (int) ellipse.getHeight());
+            return;
+        }
+    }
+
+    /**
+     * @return Returns the paint.
+     */
+    public Paint getPaint() {
+        return paint;
+    }
+
+    /**
+     * @param paint The paint to set.
+     */
+    public void setPaint(Color paint) {
+        this.paint = paint;
+    }
+
+    /**
+     * @return Returns the line style.
+     */
+    public int getLineStyle() {
+        return style;
+    }
+
+    /**
+     * @return Returns the line width.
+     */
+    public int getLineWidth() {
+        return width;
+    }
+
+    /**
+     * Sets the line style
+     * 
+     * @param lineStyle the style of the line
+     * @param lineWidth the width of the line
+     */
+    public void setStroke(int lineStyle, int lineWidth) {
```

Comment:
Confusion for me: Here you set two different properties at once. Why not having two different settes for lineStyle (Would an Enum help here btw?) and lineWidth?

Ohh, I guess lineStyle is one of ViewportGraphics.LINE_XXX stroke styles and your implementation is based on DrawShapeCommand or something familiar, don't you?

Reply:
Yes indeed